### PR TITLE
feature: Update nginx to support Patternfly (PROJQUAY-4124)

### PR DESCRIPTION
**Pull request by HammerMeetNail**
Note: There will be a follow up PR to clean up the nginx config related to setting cookies. Expect cookie values to change.  

* Removes legacy `Dockerfile.osbs`
* Updates `Dockerfile` to build new Quay UI based on Patternfly
* Adds new `make` commands for building static assets locally

**Testing Instructions**:
1. Remove all running instances of Quay
2. Run `make local-dev-up-static`
3. Wait for Quay to fully come online. This could be a few minutes after the container starts. 
4. Navigate to [http://localhost:8080/](http://localhost:8080/) to load the current UI
5. Navigate to [http://localhost:8080/react](http://localhost:8080/react) to load the new UI
6. Delete the cookie or navigate to [http://localhost:8080/angular](http://localhost:8080/angular) to revert to the old UI.

### CODE REVIEW

File: conf/nginx/server-base.conf.jnj [hunk 1]
```diff
@@ -102,7 +137,7 @@ location /cnr {
     {% endif %}
 }
 
-location /api/ {
```
Comment (syed) [inline conf/nginx/server-base.conf.jnj:105]: accidental spacing added

File: conf/nginx/server-base.conf.jnj [hunk 2]
```diff
@@ -18,7 +18,42 @@ proxy_redirect off;
...
+    }
+
+    root {{static_dir}}/patternfly/;
+    index index.html;
+
+    # Show new UI if patternfly cookie is set
+    if ($cookie_patternfly) {
+        # catch new static paths and direct to /index.html
+        # TODO: figure out how to match on $uri
+        rewrite ^(/organizations|/repositories|/signin) /index.html break;
```
Comment (syed) [inline conf/nginx/server-base.conf.jnj:47]: so what happens if I open a link in the new UI? let's say I click `<domain>/repositories/myrepo` in the new UI, will I always be redirected to `/` seems like bad UX

Comment (HammerMeetNail) [reply conf/nginx/server-base.conf.jnj:47]: Nah, you won't hit the backend. It's an internal route that should stay client side.

File: conf/nginx/server-base.conf.jnj (hunk 1 above)
Comment (HammerMeetNail) [reply conf/nginx/server-base.conf.jnj]: fixed

**Review by syed — APPROVED** [review]
LGTM

**Pull request merged**